pkg/gcpclient/client.go

```diff
@@ -217,6 +231,46 @@ func (c *gcpClient) ListComputeImages(opts ListComputeImagesOptions) (*compute.I
 	}
 	return call.Do()
 }
+func (c *gcpClient) ListComputeInstances(opts ListComputeInstancesOptions) ([]*compute.Instance, error) {
+	result := []*compute.Instance{}
+	req := c.computeClient.Instances.AggregatedList(c.projectName)
+	if len(opts.Fields) > 0 {
+		req.Fields(googleapi.Field(opts.Fields))
+	}
+	if len(opts.Filter) > 0 {
+		req = req.Filter(opts.Filter)
+	}
+	err := req.Pages(context.TODO(), func(list *compute.InstanceAggregatedList) error {
+		for _, scopedList := range list.Items {
+			for _, item := range scopedList.Instances {
+				result = append(result, item)
+			}
+		}
+		return nil
+	})
```

Comment:
I am uncomfortable with loading all of the instances into memory at once. Can we deal with each page separately instead? I don't think the size is a problem for the current use where there is a filter and only a few fields returned. However, the `ListComputeInstances` method is generic enough that I could see it used somewhere else without giving thought to the memory impact.

Reply:
Sure, we can also pass a function to handle each  compute.InstanceAggregatedList result instead of returning anything